models/shopify__order_lines.sql

```diff
@@ -1,8 +1,23 @@
+{{
+    config(
+        materialized='table' if target.type in ('bigquery', 'databricks', 'spark') else 'incremental',
+        unique_key='order_lines_unique_key',
+        incremental_strategy='delete+insert' if target.type in ('postgres', 'redshift', 'snowflake') else 'merge',
+        cluster_by=['order_line_id']
+        ) 
+}}
+
 with order_lines as (
 
-    select *
+    select 
+        *,
+        {{ dbt_utils.generate_surrogate_key(['source_relation', 'order_line_id']) }} as order_lines_unique_key
     from {{ var('shopify_order_line') }}
 
+    {% if is_incremental() %}
+    where cast(_fivetran_synced as date) >= {{ shopify.shopify_lookback(from_date="max(cast(_fivetran_synced as date))", interval=var('lookback_window', 3), datepart='day') }}
```

Comment:
Wondering why the 3 day lookback here when the other incrementals are 7 day as the default prescribes?

Reply:
It's because this model relies on `_fivetran_synced` instead of a `created_at` or `updated_at` field. `_fivetran_synced` will generally be sequential, so late arrivals will be between the most recent runs, but we would not expect to have a record added today with a `_fivetran_synced` date of last week. In this case the 3 days is a much larger buffer than what's needed mostly for safety.

For fields like `created_at` or `updated_at`, etc, those values would correspond closely with when they are actually synced, but they could very well arrive much later. We don't use `_fivetran_synced` when possible since more data could be re-synced by the connector than necessary. In the prior example, there just weren't any other dates available for incremental filtering.

Reply:
Gotcha, thanks for explaining 🙏